opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -144,6 +148,30 @@ private String processRegisterStructureEvent(Event event, ClientClassification c
         return null;
     }
 
+    private void processUpdateFamilyRegistrationEvent(Event event, boolean localEvents) {
+        if (!localEvents ) {
+            return;
+        }
+        try {
+            Location structure = structureRepository.getLocationById(event.getLocationId());
+
+            if (structure != null) {
+                Obs houseNumberObs = event.findObs(null, false, JsonForm.HOUSE_NUMBER);
+                if (houseNumberObs == null) {
+                    return;
+                } else if (structure.getProperties() != null
+                        && StringUtils.isNotEmpty(structure.getProperties().getName())
```

Comment:
Structure should still be populated if it does not have existing name i.e house number i.e
                        `StringUtils.isEmpty(structure.getProperties().getName())`